tests/vm_console_proxy_test.go

```diff
@@ -0,0 +1,177 @@
+package tests
+
+import (
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+var _ = Describe("VM Console Proxy Operand", func() {
+	var (
+		clusterRoleResource        testResource
+		clusterRoleBindingResource testResource
+		serviceAccountResource     testResource
+		serviceResource            testResource
+		deploymentResource         testResource
+		configMapResource          testResource
+	)
+
+	BeforeEach(func() {
+		expectedLabels := expectedLabelsFor("vm-console-proxy", "vm-console-proxy")
+		clusterRoleResource = testResource{
+			Name:           "vm-console-proxy",
+			Resource:       &rbac.ClusterRole{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(role *rbac.ClusterRole) {
+				role.Rules[0].Verbs = []string{"watch"}
+			},
+			EqualsFunc: func(old *rbac.ClusterRole, new *rbac.ClusterRole) bool {
+				return reflect.DeepEqual(old.Rules, new.Rules)
+			},
+		}
+		clusterRoleBindingResource = testResource{
+			Name:           "vm-console-proxy",
+			Resource:       &rbac.ClusterRoleBinding{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(roleBinding *rbac.ClusterRoleBinding) {
+				roleBinding.Subjects = nil
+			},
+			EqualsFunc: func(old *rbac.ClusterRoleBinding, new *rbac.ClusterRoleBinding) bool {
+				return reflect.DeepEqual(old.RoleRef, new.RoleRef) &&
+					reflect.DeepEqual(old.Subjects, new.Subjects)
+			},
+		}
+		serviceAccountResource = testResource{
+			Name:           "vm-console-proxy",
+			Namespace:      strategy.GetVmConsoleProxyNamespace(),
+			Resource:       &core.ServiceAccount{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(serviceAccount *core.ServiceAccount) {
+				serviceAccount.Secrets = nil
```

Comment:
I'm not sure we can do any reasonable update test for SA, because it's hard to change any of the fields for a test. What secret would we add?
Can you skip the update test for now? So these functions are not needed.

Reply:
Yep, thanks for your attention. :)